bitcoind-tests/tests/setup/test_util.rs

```diff
@@ -86,8 +86,8 @@ fn setup_keys(
     let mut x_only_keypairs = vec![];
     let mut x_only_pks = vec![];
 
-    for i in 0..n {
-        let keypair = bitcoin::secp256k1::Keypair::from_secret_key(&secp_sign, &sks[i]);
+    for sk in sks.iter().take(n) {
```

Comment:
In 3ad8c0e9615d6e9369a3ae37d66f4697d205ca33:

Pretty sure `take` is unnecessary here. You can just do `for sk in &sks`.